From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Wed, 20 Nov 2019 12:43:35 -0800
Subject: [PATCH] Force config to always be shared

We build the shared library but the last python-config that gets
installed was built unshared (see spec for details). We don't want to
have config act like the unshared build is the one to use though so
force shared in the config scripts.
---
 Misc/python-config.in    | 5 -----
 Misc/python-config.sh.in | 2 +-
 2 files changed, 1 insertion(+), 6 deletions(-)

diff --git a/Misc/python-config.in b/Misc/python-config.in
index ebd99da..28fb80f 100644
--- a/Misc/python-config.in
+++ b/Misc/python-config.in
@@ -57,11 +57,6 @@ for opt in opt_flags:
                 libs.append(libpython)
         libs.extend(getvar('LIBS').split() + getvar('SYSLIBS').split())
 
-        # add the prefix/lib/pythonX.Y/config dir, but only if there is no
-        # shared library in prefix/lib/.
-        if opt == '--ldflags':
-            if not getvar('Py_ENABLE_SHARED'):
-                libs.insert(0, '-L' + getvar('LIBPL'))
         print(' '.join(libs))
 
     elif opt == '--extension-suffix':
diff --git a/Misc/python-config.sh.in b/Misc/python-config.sh.in
index 2602fe2..f386ff8 100644
--- a/Misc/python-config.sh.in
+++ b/Misc/python-config.sh.in
@@ -46,7 +46,7 @@ LIBS_EMBED="-lpython${VERSION}${ABIFLAGS} @LIBS@ $SYSLIBS"
 BASECFLAGS="@BASECFLAGS@"
 LDLIBRARY="@LDLIBRARY@"
 OPT="@OPT@"
-PY_ENABLE_SHARED="@PY_ENABLE_SHARED@"
+PY_ENABLE_SHARED="1"
 LDVERSION="@LDVERSION@"
 LIBDEST=${prefix_real}/lib/python${VERSION}
 LIBPL=$(echo "@LIBPL@" | sed "s#$prefix#$prefix_real#")
